cpp/src/IceGrid/TraceLevels.h

```diff
@@ -5,58 +5,54 @@
 #ifndef ICE_GRID_TRACE_LEVELS_H
 #define ICE_GRID_TRACE_LEVELS_H
 
-#include <IceUtil/Shared.h>
 #include <Ice/CommunicatorF.h>
 #include <Ice/LoggerF.h>
 
 namespace IceGrid
 {
 
-class TraceLevels : public ::IceUtil::Shared
+class TraceLevels
 {
 public:
 
-    TraceLevels(const Ice::CommunicatorPtr&, const std::string&);
-    virtual ~TraceLevels();
+    TraceLevels(const std::shared_ptr<Ice::Communicator>&, const std::string&);
 
     const int admin;
-    const char* adminCat;
+    const std::string adminCat;
```

Comment:
Use string_view instead and initialize it in the header?

Reply:
Using string_view would make it harder to use with the logger as there's no implicit conversion from string_view to string.